backend/src/models/teams.rs

```diff
@@ -0,0 +1,58 @@
+use crate::models::entities::{teams, users};
+use crate::routes::teams::create_team::CreateTeamModel;
+use crate::utils::error::Error;
+use crate::utils::jwt::Claims;
+use chrono::Utc;
+use sea_orm::ActiveValue::Set;
+use sea_orm::{ActiveModelTrait, QueryFilter};
+use sea_orm::{ColumnTrait, DatabaseConnection, EntityTrait, TransactionTrait};
+
+impl teams::Model {
+    pub async fn create_team(
+        database: &DatabaseConnection,
+        create_team_json: CreateTeamModel,
+        claim_data: Claims,
+    ) -> Result<(), Error> {
+        let transaction = database.begin().await?;
```

Comment:
Here I think one transaction is not very intuitive. I'd move this line below `Error::TeamAlreadyExists` and here just pass `database`.